node/abTest/commands/finish.ts

```
@@ -0,0 +1,23 @@
+import { FindWorkspace, TestingWorkspaces } from '../../workspace/list'
+import { FinishABTestMaster, FinishABTestParams } from '../../workspace/modify'
+import { firstOrDefault } from '../../utils/firstOrDefault'
+
+export async function FinishAbTestForWorkspace(ctx: ColossusContext): Promise<void> {
+    const { vtex: { account, route: { params: { finishingWorkspace } } } } = ctx
+
+    try {
+        const workspaceName = firstOrDefault(finishingWorkspace)
+        if (!FindWorkspace(account, workspaceName, ctx)) {
+            ctx.body = `workspace "${finishingWorkspace}" doesn't exists.`,
+                ctx.status = 401
+        }
+
+        var testingWorkspaces = await TestingWorkspaces(account, ctx)
+        if (testingWorkspaces.length == 2) {
+            await FinishABTestMaster(account, ctx)
+        }
+        await FinishABTestParams(account, workspaceName, ctx)
+    } catch (err) {
+        throw new Error(err)
```

You can set the response status code when you catch the error.
`try` --> everything is ok, nothing wrong. In case of something unexpected, you can throw the error and let the catch set the status code of error.
`catch` --> in case of errors of something unexpected, log the error e set the status code if you know the reason of the error.
So if the error is Not Found, you can assume that it probably didn't find the workspace and set the response as you want.

(also remember to log the error)